classes/class3/mysimplehw/routes/cats.js

```diff
@@ -0,0 +1,103 @@
+var assert = require('assert');
+var catData = require('./catData');
+var mongojs = require('mongojs');
+var mongoURI = process.env.MONGOURI || "mongodb://localhost/test";
+
+// http://mafintosh.github.io/mongojs/
+var db = mongojs(mongoURI, ['cats']);
+
+// FIXME: should check for existence of collection rather than hard create
+db.createCollection('cats');
+
+var insert = function (req, res) {
+
+	// Create a random cat
+	var randCat = {
+		name: catData.names[Math.floor(Math.random()*catData.names.length)],
+		age: Math.ceil(Math.random()*100),
+		colors: (function() {
+			var randColors = [];
+
+			// choose 2 - 4 random colors
+			var numColors = Math.ceil(Math.random()*3) + 1;
+
+			for (var i = 0; i < numColors; i++) {
+				var color = catData.colors[Math.floor(Math.random()*catData.colors.length)];
+				randColors.push(color);
+			}
+			return randColors;
+		})()
+	};
+
+	// insert cat into db
+	db.cats.insert(randCat, function(err) {
+
+		// find recently inserted cat, and show
+		db.cats.findOne(randCat, function(err, cat) {
```

Comment:
Ensure you are checking for errors and you can just use the generated cat object to render your template after a successful save rather than querying for the data.